internal/controller/gatewayclass.go

```diff
@@ -0,0 +1,130 @@
+// Copyright Project Contour Authors
```

Comment:
I think there is starting to be enough logic in these controllers we need a way to ensure some more direct test coverage, or have some more detailed integration tests that we can be confident all the logic here is covered

Reply:
@sunjayBhatia the event handler is making things difficult to mock the controller's `Reconcile()` method. Looking at `cmd/contour/serve.go` a lot needs to happen to mock the `k8s.DynamicClientHandler`. Maybe it's better to use e2e for testing. Thoughts? cc: @stevesloka

Reply:
I can pick up the testing bit next, was going to add some since now we're adding logic into these controllers. Happy to take the initial framework bits for that.